core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -232,6 +233,7 @@ class TimeSeriesShard(val dataset: Dataset,
     * PartitionSet - access TSPartition using ingest record partition key in O(1) time.
     */
   private[memstore] final val partSet = PartitionSet.ofSize(InitialNumPartitions, ingestSchema, recordComp)
+  private[memstore] final val partSetLock = new StampedLock
```

Comment:
Can we document why a StampedLock was chosen here as opposed to any other lock?

Reply:
Will do.